### PR TITLE
[SAI-Qualify]scan Swss log and upload to kusto

### PR DESCRIPTION
**Pull request by ms-junyi**
### Description of PR

1. collect sairedis logs from different types of machines
2. analyze the logs, extract the key and value organized by json
3. ingest json file to the kusto table SwssInvocationReport

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

### CODE REVIEW

File: test_reporting/report_data_storage.py [hunk 1]
```diff
@@ -219,11 +235,14 @@ def upload_reboot_report(self, path_name: str = "", report_guid: str = "") -> No
         if "summary.json" in path_name:
             self._ingest_data(self.REBOOT_TIMING_TABLE, reboot_timing_data)
         elif "report.json" in path_name:
-             self._ingest_data(self.RAW_REBOOT_TIMING_TABLE, reboot_timing_data)
+            self._ingest_data(self.RAW_REBOOT_TIMING_TABLE, reboot_timing_data)
 
     def upload_expected_runs(self, expected_runs: List) -> None:
         self._ingest_data(self.EXPECTED_TEST_RUNS_TABLE, expected_runs)
 
+    def _upload_swss_log_file(self, swss_file):
```
Comment (richardyu-ms) [inline test_reporting/report_data_storage.py:243]: please check and add the type decalarations as other method definiations.

**Review by richardyu-ms — APPROVED** [review]

File: test_reporting/report_data_storage.py [hunk 2]
```diff
@@ -193,6 +203,13 @@ def upload_reachability_data(self, ping_output: List) -> None:
             result.update({"UTCTimestamp": ping_time})
         self._ingest_data(self.TESTBEDREACHABILITY_TABLE, ping_output)
 
+    def upload_swss_report_file(self, swss_file: str) -> None:
+        """Upload a report to the back-end data store.
+        Args:
+            file: json
```
Comment (wangxin) [inline test_reporting/report_data_storage.py:209]: docstring needs to comply with the actual argument list.

File: test_reporting/sairedis log scanner.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+**edit yaml file for your config**
+run the process
+```
+python file: sonic-mgmt(-int)/test_reporting/sai_swss_invocations.py
+python sai_swss_invocations.py --config_path <YOUR CONFIG FILE>
+```
+
+Example:
+```
+export TEST_REPORT_INGEST_KUSTO_CLUSTER="https://ingest-azphynetsonic.centralus.kusto.windows.net"
```
Comment (wangxin) [inline test_reporting/sairedis log scanner.md:96]: Is this example ingestion URL too specific?

File: test_reporting/report_data_storage.py (hunk 2 above)
Comment (ms-junyi) [reply test_reporting/report_data_storage.py]: thx, I solved it

File: test_reporting/sairedis log scanner.md (hunk 3 above)
Comment (ms-junyi) [reply test_reporting/sairedis log scanner.md]: thx, I solved it

**Review by wangxin — APPROVED** [review]

**Pull request merged**